pkg/noderesourcetopology/filter.go

```diff
@@ -44,23 +46,46 @@ func singleNUMAContainerLevelHandler(pod *v1.Pod, zones topologyv1alpha1.ZoneLis
 	// Node() != nil already verified in Filter(), which is the only public entry point
 	logNumaNodes("container handler NUMA resources", nodeInfo.Node().Name, nodes)
 
-	// We count here in the way TopologyManager is doing it, IOW we put InitContainers
-	// and normal containers in the one scope
-	for _, container := range append(pod.Spec.InitContainers, pod.Spec.Containers...) {
+	// the init containers are running SERIALLY and BEFORE the normal containers
+	// therefore, we don't need to accumulate their resources together
+	for _, initContainer := range pod.Spec.InitContainers {
+		logKey := fmt.Sprintf("%s/%s/%s", pod.Namespace, pod.Name, initContainer.Name)
+		klog.V(6).InfoS("target resources", resourceListToLoggable(logKey, initContainer.Resources.Requests)...)
+
+		_, match := resourcesAvailableInAnyNUMANodes(logKey, nodes, initContainer.Resources.Requests, qos, nodeInfo)
```

Comment:
Are you going to implement an optimal [container placement strategy](https://github.com/kubernetes-sigs/scheduler-plugins/issues/366#issuecomment-1098617120) in this PR?

Reply:
Thank you for the quick review.
The thing is that even if we'll implement an optimal placement strategy as you suggested, it is eventually for Kubelet to decide which container is about to be placed on which NUMA. 

Kubelet doesn't have this optimal placement strategy, so in the case above it will fail to schedule the pod, regardless of what the filter plugin thinks or do.

Hence, IMHO we should imitate Kubelet's behavior as much as possible in order to predict the future placements nearly as possible to reality.